packages/core/src/directive/directive.ts

```diff
@@ -20,13 +22,37 @@ export class Directive<T extends keyof DirectiveArgs = any> {
   params(): Param[] {
     if (!this.hasArgs()) return [];
 
+    if (isParamMap(this.args)) {
+      return Object.values(this.args)
+    }
+
     return Array.isArray(this.args)
+        // Assumes args is a array of objects which have a "params" method which returns Param[]
       ? this.args.reduce((r, x) => [...r, ...x.params()], [])
+       // Assumes args has a "params" method which returns Param[]
+       // @ts-expect-error
       : this.args.params();
+
+    // TODO: The assumptions above are not type safe. We need an interface that defines a params() method which returns Param[].
+    // this.args should then be contrained by a type which unites the above and ParamMap.
   }
 
   toString() {
-    return '@' + this.name
-      + (this.hasArgs() ? `(${this.args})` : '');
+    const out = '@' + this.name
+
+    if (!this.hasArgs()) {
+      return out
+    }
+
+    if (isParamMap(this.args)) {
+      return this.params().reduce((p, c, i) => {
+          if (i > 0) {
+            p += ', '
+          }
+          return p += `${c.getName()}: ${c.getValue()}`
```

Comment:
I think you misunderstood how `Param` works. Anywhere where you need param's value, you just stringify it, which will give something like `'$p1'` and you should use that in query not the value. the value is passed as [graphql variable](https://dgraph.io/docs/query-language/graphql-variables) to db request along with query string.

Reply:
thanks for the clarification. I refactored the build and stringfier code to output the param name and therefore leverage GraphQL vars and refactored tests to expect var names instead